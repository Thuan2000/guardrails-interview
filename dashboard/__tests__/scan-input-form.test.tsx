/**
 * Copyright Thuan Nguyen Â©2022
 * Good codes make the world a better place!
 */

import { render } from "@testing-library/react";
import ScanInputFormPage from "../pages/scan-input-form";
import { MockedProvider } from "@apollo/client/testing";
import renderer from "react-test-renderer";

describe("Scan Input Form Page: ", () => {
  it("+ renders without crashing", () => {
    render(
      <MockedProvider addTypename={false}>
        <ScanInputFormPage />
      </MockedProvider>
    );
  });

  it("+ matches snapshot", () => {
    const renderedComponent = renderer
      .create(
        <MockedProvider addTypename={false}>
          <ScanInputFormPage />
        </MockedProvider>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
