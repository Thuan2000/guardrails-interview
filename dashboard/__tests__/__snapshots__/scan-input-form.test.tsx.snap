// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scan Input Form Page:  + matches snapshot 1`] = `
<div
  className="ui container"
>
  <h3
    className="ui center aligned header"
  >
    Add Your Security Scan Result By Filling This Form
  </h3>
  <div
    style={
      {
        "margin": "auto",
        "width": "868px",
      }
    }
  >
    <div
      className="ui segment"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label>
            Repository Name
          </label>
          <div
            className="ui input"
          >
            <input
              aria-describedby={null}
              onChange={[Function]}
              placeholder="SDConnect"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Status
          </label>
          <div
            aria-describedby={null}
            aria-expanded={false}
            className="ui selection dropdown"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <div
              aria-atomic={true}
              aria-live="polite"
              className="divider default text"
              role="alert"
            >
              Status
            </div>
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
            <div
              className="menu transition"
            >
              <div
                aria-checked={false}
                aria-selected={true}
                className="selected item"
                onClick={[Function]}
                role="option"
                style={
                  {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Queued
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Inprogress
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Success
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Failure
                </span>
              </div>
            </div>
          </div>
        </div>
        <h4
          className="ui header"
        >
          Findings Input
        </h4>
        <div
          className="ui segment"
        >
          <button
            className="ui button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="icon icon-tabler icon-tabler-plus"
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <line
                x1={12}
                x2={12}
                y1={5}
                y2={19}
              />
              <line
                x1={5}
                x2={19}
                y1={12}
                y2={12}
              />
            </svg>
             Add Finding
          </button>
        </div>
        <div
          className="ui divider"
        />
        <button
          className="ui blue button"
          onClick={[Function]}
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
